tests/gen/schema_parser.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package gen
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/request/graphql"
+)
+
+func parseSchema(schema string) (map[string]client.CollectionDefinition, error) {
+	parser, err := graphql.NewParser()
+	if err != nil {
+		return nil, err
+	}
+	cols, err := parser.ParseSDL(context.Background(), schema)
+	if err != nil {
+		return nil, err
+	}
+	result := make(map[string]client.CollectionDefinition)
+	for i, col := range cols {
+		col.Description.ID = uint32(i)
```

Comment:
thought: I think this line is a notable risk.  I'm guessing a little bit here, but I assume you use this as the collection index - used by documents in the integration test harness to attach themselves to a particular collection.

thought: *If* we ever want this tool to be moved out of the test package this will very likely need changing, as it then likely need to fully mimic the production code used when creating collections, which this code does not currently do as far as I can see.

suggestion: I suggest trying to find a way to remove need for this line somehow.

todo: If this line (or similar) is preserved, it very definitely needs to be documented.

Reply:
the result of docs generation is a list of `GeneratedDoc`:
```go
type GeneratedDoc struct {
	ColID uint32
	JSON  string
}
```
This is basically the only reason why I need the collection ID. It's not used for anything else. If you think some other format of output would suit better our needs, let me know. We could use instead the collection's name.

Reply:
I think what you are doing is incorrect here then.  The `CollectionID` on the `Create/UpdateDoc` is poorly named and misleading.  It is not `Collection.ID`, but is the index of the collection within the test framework, at the moment it happens to (mostly) match the `Collection.ID`, but that is really a coincidence.

Name or the full struct (expensive if not using a pointer), would be better here I think.